src/inject/inject.js

```diff
@@ -1,48 +1,77 @@
-    input.value = text;
-    document.body.appendChild(input);
-    input.select();
-    document.execCommand('Copy');
-    document.body.removeChild(input);
-}
-
-try {
-    const allItemNames = Array.from(document.getElementsByClassName('item-list__list sys_item_row'))
-        .map(el => {
-            return {
-                name: el.childNodes[3].childNodes[1].innerText,
-                amount: +el.dataset.stack,
-            }
-        });
+"use strict";
 
-    let langFromUrl = /(fr|na|de|jp|eu).finalfantasyxiv.com/.exec(window.location.href)[1];
-
-    const lang = {
-        'fr': 'fr',
-        'na': 'en',
-        'eu': 'en',
-        'jp': 'ja',
-        'de': 'de'
-    }[langFromUrl];
-
-    fetch(chrome.runtime.getURL('data/items.json'))
-        .then((response) => response.json())
-        .then(items => {
-            const allIds = allItemNames.map(item => {
-                const id = Object.keys(items)
-                    .find(key => items[key][lang].trim().toLowerCase() === item.name.trim().toLowerCase());
+void (async () => {
+    const api = typeof browser === "object" ? browser : chrome;
+
+    try {
+        const allItemNames = Array.from(
+            document.getElementsByClassName("item-list__list sys_item_row"),
+            el => {
                 return {
-                    ...item,
-                    id: +id
+                    name: el.querySelector(".item-list__name--inline > a")
+                        .textContent,
+                    amount: parseInt(el.dataset.stack, 10)
                 };
-            });
-            copyToClipboard(JSON.stringify(allIds));
-            alert('Inventory copied inside your clipboard');
+            }
+        );
+
+        const langFromUrl = /(fr|na|de|jp|eu).finalfantasyxiv.com/.exec(
+            location.href
+        )[1];
+
+        const lang = {
+            fr: "fr",
+            na: "en",
+            eu: "en",
+            jp: "ja",
+            de: "de"
+        }[langFromUrl];
+
+        const items = await (await fetch(
+            api.runtime.getURL("/data/items.json")
+        )).json();
+        const nameMap = new Map(
+            Object.entries(items).map(([id, { [lang]: name }]) => [name, id])
+        );
+        const nameList = Array.from(nameMap.keys());
+        const allIds = allItemNames.map(item => {
+            const id = nameMap.get(
+                nameList.find(
+                    name => name.trim().localeCompare(item.name.trim()) === 0
+                )
+            );
+            return {
+                ...item,
+                id: id !== undefined ? parseInt(id, 10) : -1
+            };
         });
-} catch (_) {
-    alert('Please click this extension only when you\'re inside retainer\'s inventory page (https://finalfantasyxiv.com/lodestone/character/.../retainer/.../baggage)');
-}
+        await copyToClipboard(JSON.stringify(allIds));
+
+        alert("Inventory copied inside your clipboard");
```

Comment:
Might be nice to convert this to a `notification` instead, but I think that requires an additional permission scope